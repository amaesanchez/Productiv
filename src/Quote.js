import React from "react";

/** Componenent for quote -- renders text and author
 *
 * App -> QuoteApp -> Quote
 */
function Quote({ quote }) {
  return (
    <p className="QuoteText"><i>{quote.text} - {quote.author}</i></p>
  )
}

export default Quote;
