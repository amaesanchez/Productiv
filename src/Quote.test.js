import { render } from "@testing-library/react";
import Quote from "./Quote";

const TEST_QUOTE = {
  text: "Learn from yesterday, live for today, hope for tomorrow.",
  author: "Albert Einstein",
};

describe("Quote", function () {
  it("renders without crashing", function () {
    render(<Quote quote={TEST_QUOTE} />);
  });

  it("matches snapshot", function () {
    const { container } = render(<Quote quote={TEST_QUOTE} />);
    expect(container).toMatchSnapshot();
  });

  it("Renders appropriate quote on DOM", function () {
    const { queryByText } = render(<Quote quote={TEST_QUOTE} />);

    const quote = queryByText(
      `Learn from yesterday, live for today, hope for tomorrow. - Albert Einstein`
    );

    expect(quote).toBeInTheDocument();
  });
});
