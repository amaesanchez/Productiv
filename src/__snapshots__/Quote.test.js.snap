// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote matches snapshot 1`] = `
<div>
  <p
    class="quote"
  >
    <i>
      Learn from yesterday, live for today, hope for tomorrow.
       - 
      Albert Einstein
    </i>
  </p>
</div>
`;
